### PR TITLE
Extract CallContract and RegistryInfo traits into their own crate

### PR DESCRIPTION
**Pull request by HCastano**
This PR takes a stab at part of #10130.

As part of this PR `service-transaction-checker` (which depended on CallContract and RegistryInfo) got moved from `ethcore` to `ethcore-miner`.

### CODE REVIEW

File: ethcore/call-contract/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Ethereum.  If not, see <http://www.gnu.org/licenses/>.
+
+#![warn(missing_docs)]
+
+//! Call Contract module
+//! Provides CallContract and RegistryInfo traits
```
Comment (tomusdrw) [inline ethcore/call-contract/src/lib.rs]: I would add a bit more explanation, maybe something like:

```suggestion
//! 
//! This crate exposes traits required to call contracts at particular block.
//! All utilities that depend on on-chain data should use those traits to access it.
```

File: ethcore/call-contract/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+[package]
+name = "call-contract"
```
Comment (tomusdrw) [inline ethcore/call-contract/Cargo.toml]: I wonder if it should be called `ethcore-call-contract` for consistency. While I don't like `ethcore-` prefix, I think it's good to prepend our internal crates with something. Any second opinion on this?

Comment (sorpaas) [reply ethcore/call-contract/Cargo.toml]: Second this!

Comment (5chdn) [reply ethcore/call-contract/Cargo.toml]:
```suggestion
name = "ethcore-call-contract"
```

**Pull request merged**